modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs

```diff
@@ -545,6 +545,41 @@ private static bool TryDeserializeSingleDelegate(byte[] buffer, [MaybeNullWhen(f
             return type;
         }
 
+        internal static bool IsDelegateCollectible(Delegate @delegate)
+        {
+            if (@delegate is null)
+            {
+                return false;
+            }
+
+            if (@delegate is MulticastDelegate multicastDelegate)
+            {
+                Delegate[] invocationList = multicastDelegate.GetInvocationList();
+
+                if (invocationList.Length > 1)
+                {
+                    foreach (Delegate oneDelegate in invocationList)
+                        if (IsDelegateCollectible(oneDelegate))
+                            return true;
+
+                    return false;
+                }
+            }
+
+            if (@delegate.Method.IsCollectible)
+                return true;
+
+            object? target = @delegate.Target;
+
+            if (target is null)
+                return false;
+
+            if (target.GetType().IsCollectible)
+                return true;
```

Comment:
Does the `delegate.Target` type being collectible mean the delegate is also collectible even if the `delegate.Method` isn't?

Reply:
If `A.foo()` is in a non-collectible assembly and `B` in a collectible assembly inherits from `A` and doesn't override `foo`, a delegate to `B.foo()` should have a collectible target but not a collectible method. And then it should be possible to release the delegate itself (which may or may not be of a collectible type itself) and make the instance of `B` available to be collected and thus possibly unblock the unloading of `B`.

Anyways these checks generally default to releasing the target to be safe and only try to prevent the unloading from things in GodotTools